File: src/main/java/com/tira/restaurants/service/ReservationService.java
package com.tira.restaurants.service;

import java.time.LocalDate;
import java.time.LocalTime;import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Set;

import org.springframework.http.ResponseEntity;

import com.tira.restaurants.domain.Reservation;
import com.tira.restaurants.domain.Restaurant;

public interface ReservationService{
	Reservation makeReservation(Integer persons, LocalDate reservationDate, LocalTime reservationHour,
			Long idRestaurant, Long idUser);
	Set<Restaurant> getRestaurantsSortReservationsToday();
	Map<String, Object> checkReservationAvailability(Integer persons, LocalDate date, LocalTime time, Long idRestaurant);
}

File: ember-restaurants/app/services/reservation-service.js
import Ember from 'ember';

export default Ember.Service.extend({

  ajaxService: Ember.inject.service(),
  checkReservationAvailability: function(id, reservation) {
    return this.get('ajaxService').request('/checkReservationAvailability', {
      method: 'POST',
      data: JSON.stringify({
        idRestaurant: id,
        people: reservation.people,
        hour: reservation.hour,
        date: reservation.date
      })

    });
  },

  makeReservation: function(_idRestaurant, _persons,
                                    _reservationDate, _reservationHour, _idUser) {
    console.log(_reservationDate);
    return this.get('ajaxService').request('/makeReservation', {
      method: 'POST',
      data: JSON.stringify({
        idRestaurant: _idRestaurant,
        persons: _persons,
        reservationDate: _reservationDate,
        reservationHour: _reservationHour,
        idUser: _idUser
      })
    });
  }


});
